libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/util/config/ConfigManager.java

```diff
@@ -0,0 +1,463 @@
+ *  that works in tandem with {@link ConfigManager} to provide dynamically changing and efficient configurations
+ *  (especially - be aware of the slick {@link DynamicConfigDuration}). <br>
+ * <br>
+ * <h3>Note about system-properties</h3>
+ * By default - this class reflects all configs onto system-properties
+ *  (using {@link #systemSetProperty} and {@link #systemClearProperty}).
+ * Also, {@link #getConfigValue} uses the system-properties as fallback in case a property is
+ *  not set in the config-sources (using {@link #systemGetProperty}). <br>
+ * This behaviour can be disabled/changed by overriding these "system..." methods. <br>
+ * <br>
+ * <h3>Config-Values Translation</h3>
+ * All config values (originated from a config-source or from {@link #systemGetProperty})
+ *  are "translated" using {@link #translateConfigValue} (which by default does nothing). <br>
+ * <br>
+ */
+public class ConfigManager implements Closeable {
+
+    /** See {@link #addProvider}. All access must be "synchronized" ! */
+    private final List<ConfigProvider> configProviders = new LinkedList<>();
+
+    /** See {@link #addConfigSource}. All access must be "synchronized" ! */
+    private final List<ConfigProvider.ConfigSource> configSources = new LinkedList<>();
+
+    /** See {@link #registerChangeCallback}. All access must be "synchronized" ! */
+    private final Set<Runnable> changeCallbacks = new HashSet<>();
+
+    /** The current config-entries */
+    private final Map<String, ConfigEntry> previousConfigs = new ConcurrentHashMap<>();
+
+    private final Thread reloadSourcesThread;
+    private boolean isClosed = false;
+
+    /**
+     * Whenever we reload a config-source - this is incremented.
+     * At all times, we know - per config-entry - at what generation it was last encountered.
+     */
+    private long lastReloadGeneration = 0;
+
+    private static final Logger LOG = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+
+    /** Construct a NON-RELOADING config-manager */
+    public ConfigManager() {
+        // Add all standard providers.
+        init();
+
+        reloadSourcesThread = null;
+    }
+
+    /**
+     * Construct a RELOADING config-manager.
+     * The periodic reload interval is also configurable.
+     * Note that
+     */
+    public ConfigManager(String reloadConfigKey, long reloadDefaultValue, TimeUnit reloadTimeUnit) {
+        // Add all standard providers.
+        init();
+
+        // Periodically reload configs.
+        DynamicConfigDuration interval = new DynamicConfigDuration(this, reloadConfigKey, reloadDefaultValue, reloadTimeUnit);
+        reloadSourcesThread = new Thread(() -> {
+            while (!isClosed) {
+                try {
+                    interval.sleep(sleepMilliseconds -> Math.max(sleepMilliseconds, 1000L));   // never sleep less than 1 second - a safety mechanism to avoid 100% CPU loop
```

Comment:
Instead of 1000L and writing a comment here and in DynamicConfigDuration, replace it with const CONFIG_MANAGE_MIN_SLEEP declared in ServerCommonConsts and add a comment next to it if needed.

Reply:
Fixed.